server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -378,7 +378,10 @@ public void onFailure(Exception e) {
                                 null
                             ),
                             new SearchShardTask(123L, "", "", "", null, Collections.emptyMap()),
-                            result
+                            result.delegateFailure((l, r) -> {
+                                r.incRef();
```

Comment:
This kind of hack is fine, no need to do anything fancy, we never use this transport action with a future in prod.